tests/upgrades/test_discovery.py

```diff
@@ -60,5 +60,5 @@ def test_post_upgrade_fdi_version(self, target_sat, pre_upgrade_data):
         fdi_package = target_sat.execute('rpm -qa *foreman-discovery-image*').stdout
         # Note: The regular exp takes care of format digit.digit.digit or digit.digit.digit-digit in the output
         post_upgrade_version = Version(re.search(r'\d+\.\d+\.\d+(-\d+)?', fdi_package).group())
-        assert post_upgrade_version >= pre_upgrade_version
+        assert str(post_upgrade_version) >= pre_upgrade_version
```

Comment:
comparing versions here make more sense here, so just would like to know why this change?

Reply:
Since we dictionary doesn't accept Version so changing it to str.

Reply:
@shweta83 I didn't get, but as we discussed in our call we agreed to keep Version as is instead of comparing strings as its more effective

Reply:
Dictionary saved in pre_upgrade scenario doesn't accept Version object so I tried a new workaround for it.